src/cookie-io.js

```diff
@@ -174,12 +174,14 @@ class CharacterData {
         }
     }
     setQuestCompleted(scene) {
-        if (this.checkScene(scene)) {
-            this.characterData.levels[scene].questCompleted = true;
+        if (this.checkScene(scene.currentLevel)) {
+            this.characterData.levels[scene.currentLevel].questCompleted = true;
+
+            // Unlock levels
             for (var level in this.characterData.levels) {
                 if (
                     !this.characterData.levels[level].unlocked &&
-                    CONSTANTS.PREREQUISITES[level] == scene
+                    CONSTANTS.PREREQUISITES[level] == scene.currentLevel
```

Comment:
Why pass in the whole scene object just to use the current level name like before? If you think `scene` is confusing because we usually use it to mean the object not the string, I think renaming the input to `sceneName` would be better.

Reply:
At one point I was using another attribute of scene and I needed to pass in the object. [Reverted](https://github.com/EthanVieira/rs-clicker/pull/47/commits/0d1df380ee541e72d588204b3ac45a7aa2cea81e])!